3.0/Dockerfile

```diff
@@ -6,8 +6,7 @@
 # specific reason to use a different one. This means January, April, July, or
 # October.
 
-# NOTE: Ruby under 3.1 require OpenSSL >= 1.0.1, < 3.0.0, so use Ubuntu 20.04.
-FROM cimg/base:2023.01-20.04
+FROM cimg/base:2023.07
```

Comment:
3.0 does not support OpenSSL 3

Reply:
@JalexChen This includes a workaround that installs older openssl packages from Focal on Jammy which seems to work fine, therefore we don't need to use a Focal based image anymore. In testing it has worked fine for me, but if we see users having issues then we can re-roll 3.0 on Focal